### PR TITLE
feat: Select default value control

### LINKED ISSUE #16078: [Bug]: Select and Multiselect shows error for defaultValue even when there's none
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When I make my options from a query and make the defaultValue the first item in the query array, I get errors telling me the value isn't in the option

### Steps To Reproduce

https://user-images.githubusercontent.com/31691737/183900676-7b5fe86d-231a-47a6-a9ba-e808488c719c.mov

### Public Sample App

_No response_

### Version

Cloud

### PR DESCRIPTION
**Pull request by Tooluloope**
## Description

Current fix to default value validation dependency in Select widgets

Fixes #16078

This PR fixes the Below issue
https://user-images.githubusercontent.com/31691737/183900676-7b5fe86d-231a-47a6-a9ba-e808488c719c.mov

The correct validation can be found here 
https://github.com/appsmithorg/appsmith/issues/13664#issuecomment-1120814337

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/widgets/MultiSelectWidgetV2/widget/index.tsx [hunk 1]
```diff
@@ -144,6 +144,19 @@ export function defaultOptionValueValidation(
   }
 
   if (isValid && !_.isNil(parsed) && !_.isEmpty(parsed)) {
+    if (!Array.isArray(options) && typeof options === "string") {
```
[the remark this answers is not in the document]
Comment (Tooluloope) [reply app/client/src/widgets/MultiSelectWidgetV2/widget/index.tsx:147]: I was expecting the other branch to have been merged then I'll pull it in here, but failing test hasn't allowed me merge the branch.

**Review by sbalaji1192 — APPROVED** [review]

**Comment by chandannkumar** [comment]
There is no immediate error message popups when Default value is removed/changed from Options of Select & Multiselect widget

https://www.loom.com/share/960253da78114a988cb11346de860892

@Tooluloope

**Comment by chandannkumar** [comment]
@Tooluloope 
Added 1 more callout

**Comment by dilippitchika** [comment]
@Tooluloope why is this PR still open?

**Comment by Tooluloope** [comment]
@chandannkumar Please could you open a new Issue for No 1,
For No 2 I think you might need to set your Query better to fix it.
The filtering for the Widget is done on the server and not in the widget

**Review by sbalaji1192 — COMMENTED** [review]
Can we add some test cases to validate the hack?

**Comment by chandannkumar** [comment]
@Tooluloope 
On copying select & Multiselect widget, the default values seems incomplete in UI but default value works fine without any error

https://www.loom.com/share/0cbfbe17f3494a26a0a060d231518147

File: app/client/src/sagas/WidgetOperationSagas.tsx [hunk 2]
```diff
@@ -1458,6 +1458,27 @@ function* pasteWidgetSaga(
             }
           }
 
+          // Update the Select widget defaultValue properties
```
Comment (sbalaji1192) [inline app/client/src/sagas/WidgetOperationSagas.tsx:1461]: Could you add a TODO here to move this to the widget definition? Later we can create a static function on each widget `onPaste`, which would do this transformation on the given widget.

**Review by sbalaji1192 — COMMENTED** [review]
Could we add some cypress case to check the fix?

**Review by sbalaji1192 — APPROVED** [review]

**Comment by chandannkumar** [comment]
- Tested this PR on Select & Multi select widget
- Tested this on older apps as well

Tested this PR and working as expected

**Pull request merged**